.github/workflows/reference_linter.yaml

```
@@ -0,0 +1,32 @@
+name: Lint Reference Files
+on:
+  push:
+    paths:
+      - 'l10n/en-US/**.ftl'
+      - '.github/workflows/reference_linter.yaml'
+    branches:
+      - master
+  pull_request:
+    paths:
+      - 'l10n/en-US/**.ftl'
+      - '.github/workflows/reference_linter.yaml'
+    branches:
+      - master
+  workflow_dispatch:
+jobs:
+  linter:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Clone repository
+        uses: actions/checkout@v3
```

The other workflows were recently updated to use version 4 of this, so can we please be consistent here?
```suggestion
        uses: actions/checkout@v4
```